sklearn/metrics/pairwise.py

```diff
@@ -1716,6 +1725,19 @@ def pairwise_distances(X, Y=None, metric="euclidean", *, n_jobs=None,
         .. versionchanged:: 0.23
            Accepts `pd.NA` and converts it into `np.nan`
 
+    dtype : str, type, list of types, default=None
```

Comment:
May I ask why this is going to be "float by default" instead of "preserve the original type"? Wouldn't that require passing the dtype in each estimator that uses a metric, if you want to use it with non-numerical data?